packages/liveblocks-core/src/connection.ts

```diff
@@ -412,54 +418,103 @@ function createConnectionStateMachine<T extends BaseAuthResult>(
     .onEnterAsync(
       "@connecting.busy",
 
-      (ctx) => {
-        //
-        // Use the "connect" delegate to create the WebSocket connection (which
-        // will initiate the connection), and set up all the necessary event
-        // listeners, then wait until the 'open' event has fired.
-        //
-        const promise = new Promise<IWebSocketInstance>((res, rej) => {
-          if (ctx.token === null) {
-            throw new Error("No auth token"); // This should never happen
-          }
+      //
+      // Use the "createSocket" delegate function (provided to the
+      // ManagedSocket) to create the actual WebSocket connection instance.
+      // Then, set up all the necessary event listeners, and wait for the
+      // "open" event to occur.
+      //
+      // When the "open" event happens, we're ready to transition to the
+      // OK state. This is done by resolving the Promise.
+      //
+      async (ctx) => {
+        let capturedPrematureEvent: IWebSocketEvent | null = null;
+
+        const promise = new Promise<[IWebSocketInstance, () => void]>(
```

Comment:
I know that 90% of my critiques are about naming things, but can we call this something other than promise?

Reply:
Fixed in e4060b6f8c63247cfbe3986e00511d6a74f85b62. I've renamed it to `connect$`, using the convention of naming promises (values that still need to be `await`'ed) with a `$`-suffix, which we also do elsewhere in code to make that more explicit.